pkg/frontend/querymiddleware/split_and_cache.go

```diff
@@ -163,6 +163,7 @@ func (s *splitAndCacheMiddleware) Do(ctx context.Context, req Request) (Response
 		for _, splitReq := range splitReqs {
 			// Do not try to pick response from cache at all if the request is not cachable.
 			if cachable, reason := isRequestCachable(splitReq.orig, maxCacheTime, cacheUnalignedRequests, s.logger); !cachable {
+				level.Debug(s.logger).Log("msg", "skipping response cache as query is not cacheable", "query", splitReq, "reason", reason, "tenants", tenant.JoinTenantIDs(tenantIDs))
```

Comment:
`splitReq` isn't a query, it's a struct and so you won't be able to log its value here. I think what you want is `splitReq.orig.GetQuery()`